dmapiclient/base.py

```diff
@@ -113,7 +113,12 @@ def _request(self, method, url, data=None, params=None):
             "Authorization": "Bearer {}".format(self.auth_token),
             "User-agent": "DM-API-Client/{}".format(__version__),
         }
-        headers = self._add_request_id_header(headers)
+        if has_request_context():
+            if callable(getattr(request, "get_onwards_request_headers", None)):
+                headers.update(request.get_onwards_request_headers())
+            elif getattr(request, "request_id", None) and current_app.config.get("DM_REQUEST_ID_HEADER"):
+                # support old .request_id attr for compatibility
```

Comment:
I wonder if worth putting this as a TODO to remove? Or will you be doing that very imminently?

Reply:
Hopefully the latter, but would be good to do a collective think over whether I've missed anywhere...